src/configuration.ts

```diff
@@ -216,14 +237,14 @@ export class Configuration {
     this._serviceContext.version = isString(version) ? version : undefined;
 
     if (isObject(this._givenConfiguration.serviceContext)) {
-      if (isString(this._givenConfiguration.serviceContext.service)) {
-        this._serviceContext.service = this._givenConfiguration.serviceContext.service;
+      if (isString(this._givenConfiguration.serviceContext!.service)) {
```

Comment:
Can we really assume that `serviceContext` is never null or undefined here? I don't know what exactly `isObject` does.

Reply:
`isObject` returns `false` for `null` and `undefined` and so the use of `!` is fine.  As follow-up work, I want to update this code so it is not using `is*` methods because we won't need them anymore with type checking.  For now, I'm using `!` to manually tell Typescript the value is defined.